lib/test_prof/factory_default.rb

```diff
@@ -29,17 +32,26 @@ def init
         TestProf::FactoryBot::Strategy::Stub.prepend StrategyExt
 
         @store = {}
+        @store_preserve_traits = {}
+        # default is false to retain backward compatibility
+        @preserve_traits = false
       end
 
-      def register(name, obj)
+      def register(name, obj, preserve_traits = nil)
+        unless FactoryDefault.preserve_traits
+          @store_preserve_traits[name] ||= true if preserve_traits
+        end
         store[name] = obj
```

Comment:
What about storing the options along with the object itself, i.e.:

```ruby
store[name] = { record: obj, **options}
```

Reply:
First thought was to save some hash allocations, but agree, no point in chasing those several dozens additional hashes, also if we decide to make some default-for-trait-objects they can be stored in the same hash under special keys.